lib/LTP/WhiteList.pm

```diff
@@ -19,23 +18,28 @@ use Mojo::UserAgent;
 use Mojo::File 'path';
 use YAML::PP;
 
-our @EXPORT = qw(
-  download_whitelist
-  find_whitelist_testsuite
-  find_whitelist_entry
-  override_known_failures
-  is_test_disabled
-  list_skipped_tests
-);
+sub new {
+    my $class = shift;
+    my $self = bless({}, $class);
+    my $path = get_var('LTP_KNOWN_ISSUES_LOCAL');
+
+    if (!defined($path) && get_var('LTP_KNOWN_ISSUES')) {
+        $path = _download_whitelist();
+    }
 
-sub download_whitelist {
+    $self->{whitelist} = _load_whitelist_file($path) if $path;
+    $self->{whitelist} ||= {};
+    return $self;
+}
```

Comment:
i was thinking that with Mojo::Base, you could avoid some boilerplate. WDYT?

Reply:
Can you be more specific?

Reply:
After clarifying this a bit on IRC, I'm not going to use any of those special features. It'd add nothing but unnecessary complexity.